corehq/form_processor/backends/couch/dbaccessors.py

```diff
@@ -126,6 +131,14 @@ def _form_undelete(doc):
             doc['server_modified_on'] = json_format_datetime(datetime.utcnow())
         return _soft_undelete(XFormInstance.get_db(), form_ids, _form_undelete)
 
+    @staticmethod
+    def modify_attachment_xml_and_metadata(form_data, form_attachment_new_xml):
+        form_data.put_attachment(form_attachment_new_xml, name="form.xml", content_type='text/xml')
+        operation = XFormOperation(user_id=SYSTEM_USER_ID, date=datetime.utcnow(),
+                                   operation='Scrub username for GDPR compliance.')
+        form_data.history.append(operation)  # TODO: should this show in Form History tab? it doesn't
```

Comment:
I think this TODO should be resolved before the PR is merged if possible. What do you think?

Reply:
check out this comment to see how you might achieve this: https://github.com/dimagi/commcare-hq/pull/19973#discussion_r177989201

Reply:
This is operating on the couch models so this approach should be fine. A unit test should be easy to make sure it's working

Reply:
Tested here:
https://github.com/dimagi/commcare-hq/pull/19979/commits/8f4cf33e4b0319561c0110c5bb470e9790b3bb27